src/inc/til/rle.h

```diff
@@ -0,0 +1,1066 @@
+            }
+        }
+
+        // Places this value in every position from start to end.
+        // If no start is specified, fills the entire list.
+        void fill(const T value, const S start = gsl::narrow_cast<S>(0))
+        {
+            insert(value, start, gsl::narrow_cast<S>(_size - start));
+        }
+
+        // Inserts this value at the given position for the given run length.
+        void insert(const T value, const S position, const S length = gsl::narrow_cast<S>(1))
+        {
+            // TODO: validate position in bounds?
+            std::pair<T, S> item{ value, length };
+            gsl::span<std::pair<T, S>> span{ &item, 1 };
+            _merge(span.begin(), span.end(), position);
+        }
+
+        // Inserts all values between first and last starting at the given position.
+        template<class Iter>
+        void assign(Iter first, Iter last, const S position = gsl::narrow_cast<S>(0))
+        {
+            //TODO: validate that it doesn't overrun?
+            _merge(first, last, position);
+        }
+
+        constexpr bool operator==(const rle& other) const noexcept
+        {
+            return _size == other._size &&
+                   std::equal(_list.cbegin(), _list.cend(), other._list.cbegin());
+        }
+
+        constexpr bool operator!=(const rle& other) const noexcept
+        {
+            return !(*this == other);
+        }
+
+        /*[[nodiscard]] iterator begin() noexcept
+        {
+            return iterator(_list.begin());
+        }*/
+
+        [[nodiscard]] const_iterator begin() const noexcept
+        {
+            return const_iterator(_list.begin());
+        }
+
+        /*[[nodiscard]] iterator end() noexcept
+        {
+            return iterator(_list.end());
+        }*/
+
+        [[nodiscard]] const_iterator end() const noexcept
+        {
+            return const_iterator(_list.end());
+        }
+
+        /*[[nodiscard]] reverse_iterator rbegin() noexcept
+        {
+            return reverse_iterator(end());
+        }*/
+
+        [[nodiscard]] const_reverse_iterator rbegin() const noexcept
+        {
+            return const_reverse_iterator(end());
+        }
+
+        /*[[nodiscard]] reverse_iterator rend() noexcept
+        {
+            return reverse_iterator(begin());
+        }*/
+
+        [[nodiscard]] const_reverse_iterator rend() const noexcept
+        {
+            return const_reverse_iterator(begin());
+        }
+
+        [[nodiscard]] const_iterator cbegin() const noexcept
+        {
+            return begin();
+        }
+
+        [[nodiscard]] const_iterator cend() const noexcept
+        {
+            return end();
+        }
+
+        [[nodiscard]] const_reverse_iterator crbegin() const noexcept
+        {
+            return rbegin();
+        }
+
+        [[nodiscard]] const_reverse_iterator crend() const noexcept
+        {
+            return rend();
+        }
+
+        std::wstring to_string() const

```

Comment:
Maybe `std::string` is enough? Since this method isn't going to be used for things other than debugging, right?

Reply:
TAEF deals better with wstring than string. And I could honestly ifdef this out to DBG only since it's only purpose is for the TAEF output on error. So I'll likely do that.